packages/ref-core/src/ref_core/metrics.py

```diff
@@ -87,6 +91,79 @@ class TriggerInfo:
     # dataset metadata
 
 
+@frozen(hash=True)
+class DataRequirement:
+    """
+    Definition of the input datasets that a metric requires to run.
+
+    This is used to determine the different metric executions that are required to be performed
+    with the current data catalog.
+
+    The data catalog is filtered according to the `filters` field,
+    then grouped according to the `group_by` field,
+    and then each group is checked that it satisfies the `constraints`.
+    Each such group will be processed as a separate execution of the metric.
+    """
+
+    source_type: SourceDatasetType
```

Comment:
Isn't this just another facet? Or is it treated separately because you plan to validate the facet names themselves at some point?

Reply:
I think the answer is maybe. I could imagine non-CMIP data being filtered in a similar way e.g. previous executions of the metrics. I'm not certain if we want to lump these different data catalogs together or keep them separate. They will have quite different facets.

If we opt to split the requirements by type (https://github.com/CMIP-REF/cmip-ref/pull/15#discussion_r1849585098) then this field is superfluous.